lib/os/shutdown.c

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/shutdown_internal.h>
+#include <zephyr/toolchain.h>
+
+#if defined(CONFIG_MULTITHREADING) && !defined(CONFIG_SMP)
+#include <zephyr/kernel.h>
+#endif
+
+void sys_shutdown(void)
+{
+#if defined(CONFIG_MULTITHREADING) && !defined(CONFIG_SMP)
```

Comment:
I think I finally understand what @andyross meant and I agree that the inconsistency between SMP and single-processor is not something we want to introduce. On top of that, `sys_reboot()` is fully synchronous and immediate, so I would vote for this API to become a synchronous, immediate one as well for consistency.